src/main/java/com/jnape/palatable/shoki/impl/HashMultiSet.java

```diff
@@ -35,12 +37,21 @@
  */
 public final class HashMultiSet<A> implements MultiSet<A> {
 
-    private static final HashMultiSet<?> EMPTY_OBJECT_DEFAULTS = new HashMultiSet<>(HashMap.empty());
+    private static final HashMultiSet<?> EMPTY_OBJECT_DEFAULTS = new HashMultiSet<>(
+            HashMap.empty(EquivalenceRelation.objectEquals(), HashingAlgorithm.objectHashCode()),
+            EquivalenceRelation.objectEquals(),
+            HashingAlgorithm.objectHashCode());
 
-    private final HashMap<A, NonZero> multiplicityMap;
+    private final HashMap<A, NonZero>    multiplicityMap;
+    private final EquivalenceRelation<A> equivalenceRelation;
+    private final HashingAlgorithm<A>    hashingAlgorithm;
 
-    private HashMultiSet(HashMap<A, NonZero> multiplicityMap) {
-        this.multiplicityMap = multiplicityMap;
+    private HashMultiSet(HashMap<A, NonZero> multiplicityMap,
+                         EquivalenceRelation<A> equivalenceRelation,
+                         HashingAlgorithm<A> hashingAlgorithm) {
+        this.multiplicityMap     = multiplicityMap;
+        this.equivalenceRelation = equivalenceRelation;
+        this.hashingAlgorithm    = hashingAlgorithm;
```

Comment:
I would rather not have this but I'm not sure what else to do for `HashMultiSet#unique` other than being able to get the `EquivalenceRelation`+`HashingAlgorithm` off of the `HashMap`